### PR TITLE
feat(tidb): support fetch verify artifact

### PR DESCRIPTION
**Pull request by purelind**
Currently, we obtain the latest compiled binary for the corresponding branch. This approach has a hidden risk as dependencies can cause issues that affect the current pipeline.
This PR support fetch artifact binary that already passed some e2e tests to avoid the above-mentioned problems.

### CODE REVIEW

File: libraries/tipipeline/vars/component.groovy [hunk 1]
```diff
@@ -96,18 +96,20 @@ def checkoutV2(gitUrl, keyInComment, prTargetBranch, prCommentBody, credentialsI
...
 // fetch component artifact from artifactory(current http server)
-def fetchAndExtractArtifact(serverUrl, keyInComment, prTargetBranch, prCommentBody, artifactPath, pathInArchive="", trunkBranch="master") {
-    def componentBranch = computeBranchFromPR(keyInComment, prTargetBranch, prCommentBody,  trunkBranch)
-
+def fetchAndExtractArtifact(serverUrl, keyInComment, prTargetBranch, prCommentBody, artifactPath, pathInArchive="", trunkBranch="master", artifactVerify=false) {
+    def componentBranch = computeBranchFromPR(keyInComment, prTargetBranch, prCommentBody, trunkBranch)
+    def refUrl = "${serverUrl}/download/refs/pingcap/${keyInComment}/${componentBranch}/sha1"
+    if (artifactVerify) {
+        refUrl = "${serverUrl}/download/refs/pingcap/${keyInComment}/${componentBranch}/sha1.verify"
+    }
```
Comment (wuhuizuo) [inline libraries/tipipeline/vars/component.groovy:104]: suggest keep shell style rather than groovy style.

**Comment by purelind** [comment]
test passed
* https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fghpr_check2/detail/ghpr_check2/32115/pipeline
* https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fghpr_check2/detail/ghpr_check2/32114/pipeline/54/

File: libraries/tipipeline/vars/component.groovy (hunk 1 above)
Comment (purelind) [reply libraries/tipipeline/vars/component.groovy]: done

**Review by wuhuizuo — APPROVED** [review]

**Pull request merged**